src/mbgl/map/map.cpp

```diff
@@ -364,6 +364,21 @@ LatLng Map::latLngForPixel(const ScreenCoordinate& pixel) const {
     return impl->transform.screenCoordinateToLatLng(pixel);
 }
 
+std::vector<ScreenCoordinate> Map::pixelsForLatLngs(const std::vector<LatLng>& latLngs) const {
+    std::vector<LatLng> unwrappedLatLngs;
+    unwrappedLatLngs.reserve(latLngs.size());
+    for (const auto& latLng : latLngs) {
+        LatLng unwrappedLatLng = latLng.wrapped();
+        unwrappedLatLng.unwrapForShortestPath(impl->transform.getLatLng());
+        unwrappedLatLngs.emplace_back(unwrappedLatLng);
+    }
```

Comment:
why not just
```
std::vector<ScreenCoordinate> result;
result.reserve(latLngs.size())
for (const auto& latLng : latLngs)  result.emplace_back(pixelForLatLng(latLng));
return result;
```

Reply:
It is a nested function call, I do this way in purpose because by moving loops inside function instead of looping from the highest functions, it could save more function stack frame creations and also to avoid multiple memory allocation/deallocations for mat4 in the last function call.

Reply:
@zmiao you can quickly add benchmark test for two variants and measure diff between your simple variant that Mikhail proposed.

Reply:
Agreed with @pozdnyakov and @alexshalamov , for this pr, we take the simple approach. Since the most perfomance serious part is inside `TransformState`, and we plan to take it as a follow-up action of refactoring this class after this pr is landed.  Two simple benchmark tests are added, which could be used later for the performance comparison.